log/zapcore_handler.go

```diff
@@ -0,0 +1,123 @@
+package log
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"istio.io/pkg/structured"
+)
+
+var (
+	toLevel = map[zapcore.Level]Level{
+		zapcore.FatalLevel: FatalLevel,
+		zapcore.ErrorLevel: ErrorLevel,
+		zapcore.WarnLevel:  WarnLevel,
+		zapcore.InfoLevel:  InfoLevel,
+		zapcore.DebugLevel: DebugLevel,
+	}
+	toZapLevel = map[Level]zapcore.Level{
+		FatalLevel: zapcore.FatalLevel,
+		ErrorLevel: zapcore.ErrorLevel,
+		WarnLevel:  zapcore.WarnLevel,
+		InfoLevel:  zapcore.InfoLevel,
+		DebugLevel: zapcore.DebugLevel,
+	}
+)
+
+func init() {
+	RegisterDefaultHandler(ZapLogHandlerCallbackFunc)
+}
+
+// ZapLogHandlerCallbackFunc is the handler function that emulates the previous Istio logging output and adds
+// support for errdict package and labels logging.
+func ZapLogHandlerCallbackFunc(
+	level Level,
+	scope *Scope,
+	ie *structured.Error,
+	msg string,
+	fields []zapcore.Field) {
+	if ie != nil {
+		fields = append(fields, zap.String("message", msg))
```

Comment:
We currently have a `--log_as_json` option. What happens when this is enabled and we have structured logging? hopefully not double nested json?

Reply:
As a further point, does it make sense to output json only if that flag is enabled, and if its disabled something like `Hello moreInfo=some more info impact=some impact action=some action likelyCauses=some cause`? Similar to https://github.com/sirupsen/logrus format I guess

Reply:
yes, will make this change in a bit.

Reply:
done.